packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1783,6 +1786,47 @@ function updateEffect(
   return updateEffectImpl(PassiveEffect, HookPassive, create, deps);
 }
 
+function mountEvent<T>(callback: () => T): () => T {
+  const hook = mountWorkInProgressHook();
+  const ref = {current: callback};
+
+  function event(...args) {
+    if (isAlreadyRenderingProd()) {
+      throw new Error('An event from useEvent was called during render.');
+    }
+    return ref.current.apply(this, args);
```

Comment:
Is there a test that the `this` context is preserved? Probably merits one

Reply:
Or maybe we shouldn't support this and should always call with `null`.

Reply:
Yeah as long as there’s a test because once it’s out there someone will surely depend on it either way